docs/proposals/202012_receive_split.md

```diff
@@ -8,30 +8,79 @@ status: accepted
+
 
 ### Goals
 
+* Reduce the downtime of ingestion part of Thanos
+
+### Proposal
+
+We propose allowing to run Thanos Receive in a mode that only forwards/replicates remote write (distributor mode). You can enable that mode by simply
+not specifying:
+
+```yaml
+  --receive.local-endpoint=RECEIVE.LOCAL-ENDPOINT
+                                 Endpoint of local receive node. Used to
+                                 identify the local node in the hashring
+                                 configuration.
+```
+
+We can call this mode a "Routing Receive". Similarly, we can skip specify any hashring to Thanos Receive (`--receive.hashrings-file=<path>`), explicitly purposing it only for ingesting. We can call this mode "Ingesting Receive".
+
+User can also mix all of those two modes for various federated hashrings etc. So instead of what we had before:
+
+![Before](https://docs.google.com/drawings/d/e/2PACX-1vTfko27YB_3ab7ZL8ODNG5uCcrpqKxhmqaz3lW-yhGN3_oNxkTrqXmwwlcZjaWf3cGgAJIM4CMwwkEV/pub?w=960&h=720)
+
+We have:
+
+![After](https://docs.google.com/drawings/d/e/2PACX-1vTVrtCGjR4iMbrU7Kj6QAn1a1m4fr-kvoQVDAK4lzQ_wWfXfpLLEE9HB948-WHI5ZG6s1iGWt51R593/pub?w=960&h=720)
+
+This allows us to (optionally) model deployment in a way that avoid expensive re-configuration of the stateful ingesting Receives after the hashring configuration file has changed.
+
+In comparison to previous proposal (as mentioned in [alternatives](#previous-proposal-separate-receive-route-command) we have big adventages:
+
+1. We can reduce number of components in Thanos system, we can reuse similar component flags and documentation. Users has to learn about one less command and in result Thanos design is much more approachable. Less components mean less maintainance, code and other implicit duties: Separate changelogs, issue confusions, boilerplates, etc.
+2. Allow consistent pattern with Query. We don't have separate StorAPI component for proxying, we have that baked into Querier. This has been proven to be flexible and understandable, so I would like to propose similar pattern in Receive.
+3. This is more future proof for potential advanced cases like chain of routers -> receives -> routers -> receives for federated writes.
+
+### Open Questions
+
+* Buffering the incoming requests.
+
+### Plan
+
+* Receive without `--receive.hashrings` does not forward or replicate requests, it routes straight to multi-tsdb.
```

Comment:
So by **_not using a flag_** we enforce a certain behaviour, shouldn't we use a flag to enforce what we want?

Reply:
Well that's the thing. If you don't use hashring flag, it's by design impossible to do hashring forwarding, because you did not specify hashring, right? (:  That's why we can use it as quite explicit turn off of forwarding logic, no?